src/main/java/graphql/Assert.java

```diff
@@ -76,6 +78,13 @@ public static void assertTrue(boolean condition, String format, Object... args)
         throw new AssertException(format(format, args));
     }
 
+    public static void assertTrueLazily(boolean condition, String format, Supplier<?>... args) {
+        if (condition) {
+            return;
+        }
+        throw new AssertException(format(format, Arrays.stream(args).map(Supplier::get).toArray()));
+    }
+
```

Comment:
I wonder if its just a string that is laziyl supplied and the caller does the formatting


```
    assertTrue(x == y, () -> String.format("This is the message %s, sArg1));
```

this way we can re-use the simple assertTrue name and migrate to it in more places

Reply:
yes lets do it this way @bbakerman